### PR TITLE
Add rule for multiline enum cases

### PR DESCRIPTION
**Pull request by facumenzella**
### Why?

Reviewing changes within big enum declarations can be quite messy and hard to follow:

```diff
enum Enum1: Int {
-   case a = 0, p = 2, c, d
+   case a = 0, p = 4, c, f
    case e, k
    case m(String, String)
}
```

### Rule

This PR adds a rule to rewrite enum declaration cases into one case per line. As you can see in the added test, it will go from this:

```swift
enum Enum1: Int {
     case a = 0, p = 2, c, d
     case e, k
     case m(String, String)
}
```

to this:

```swift
enum Enum1: Int {
     case a = 0
     case p = 2
     case c
     case d
     case e
     case k
     case m(String, String)
}
```

Improving the visibility of further changes in the enum declaration

```diff
enum Enum1: Int {
     case a = 0
-    case p = 2
+    case p = 4
     case c
     case d
-    case e
+    case f
     case k
     case m(String, String)
```

### Improvements
- The name of the rule can probably be improved.
- Sort the cases
- Extend the rule to handle switch cases.

### CODE REVIEW

**Comment by nicklockwood** [comment]
@facumenzella thanks, this looks useful. Since it's a new rule I'll defer including it until the next major version (0.46.0).

File: Rules.md [hunk 1]
```diff
@@ -610,6 +611,10 @@ Option | Description
 </details>
 <br/>
 
+## multilineEnumCases
+
+Sort switch cases alphabetically.
```
Comment (nicklockwood) [inline Rules.md:616]: This description doesn't seem correct.

Comment (facumenzella) [reply Rules.md:616]: Sorry 😅 
I started with that but then I thought that it was probably a better idea to start simpler. I'll update this.

Comment (nicklockwood) [reply Rules.md:616]: Note that the Rules.md file is generated automatically by a unit test from the help parameter in the rule itself, so you probably just need to run the tests again to fix this rather than editing manually.

**Comment by nicklockwood** [comment]
> Sort the cases
> Extend the rule to handle switch cases.

These should be separate rules.
